### DISCUSSION TITLE
IBM Acquires Hashi

### DISCUSSION THREAD

**Discussion by jmls** [openbao/openbao#304] [General]
https://www.forbes.com/sites/stevemcdowell/2024/04/24/ibm-bolsters-hybrid-cloud-business-with-64b-hashicorp-acquisition/

I wonder how this is going to pan out .. do you think IBM will open up vault, reverse the licence change ? How is it going to affect this project and opentofu ?

So many questions

**Comment by cipherboy** [comment]
Someone asked similar on the Matrix bridge when the news first leaked; my response there was:

> I'm not sure anyone here can answer that, especially at this stage. The details of the acquisition have not been made public yet (and are still likely being negotiated -- it seems this report was from a leak), and indeed, the acquisition will take many months to complete even if all goes well (Red Hat took 9 months or so). Given HashiCorp has not reverted to the MPL in the face of OpenTofu, I would suspect we are unlikely to see any announcement or decision about the MPL until after the acquisition is complete, if and when it is successful.
> 
> Regardless, this fork is the Linux Foundation's, not IBM's; Jan and I and others are not paid by IBM and there are non-IBM TSC members. IMO, in the interim, our best option is to continue contributing to OpenBao, and if Vault were to be made MPL again, we can figure out how to combine the projects. :-)

Since then, the acquisition has been made public, but not many more details. I suspect we'll have to wait; these things take time due to regulatory approvals and other things, and until then, the businesses usually have to operate as if nothing changed.

In light of that, OpenBao will likely continue as usual. For myself, I've just been rather busy as of late and not had as much time as I would've hoped to contribute. :-)

**Reply by vtrenton** [reply]
> Red Hat took 9 months or so

The first thing Red Hat did under IBM is kill CentOS and begin to close source RHEL (not legally persay). I'm wondering why there is an asumption that IBM would revert the changes of Hashicorp. IBM alone has a lot of proprietary software they've made over the years. I'm wondering what - if any - Incentive IBM would have to Open Source Vault again? To me it seems unlikely.

**Reply by cipherboy** [reply]
@vtrenton Hmmm... My memory of the event, having worked at Red Hat at the time it was announced, was that, while killing CentOS was a deeply unpopular move internally with Engineering, it came from within and not from IBM. Red Hat, to this day, remains an independent subsidiary, with separate e.g., benefits & policies than its parent company. I'm not passing judgement on the decision, just to state that I don't think it is entirely fair to pin the blame on IBM directly for the idea.

The incentive, IMO, is this community. As long as it continues to exist and people derive benefit and contribute back (whether that means code, documentation, filing bug reports, showing up and presenting at conferences, writing blog articles, &c -- there are many ways to contribute positively), it will continue to exist under the Linux Foundation and thus be a detriment to the newly acquired business. The benefits of re-opening Vault are that the innovative work being done in this community (albeit, slowly as nobody is paid to work on it to my knowledge), can be shared by Vault again. And, if done carefully (bringing the Vault assets under the LF and giving HashiCorp a seat on the TSC more directly than via the existing IBM seat), it can ultimately be done safely for all parties involved, IMHO.

Further, IBM (or some portion thereof) _must_ recognize the benefits of the LF and the open source community, as they were the ones that started this fork under the LF in the first place. :-) They took the time and money to set up the OpenBao foundation, solicited members to co-found the TSC (mostly coming from their OpenHorizon community), and continue to market it (#148 / #236) e.g., at the LF Edge ONE summit. That would be a lot of work for some acquisition negotiations (if the fork was even started prior to acquisition negotiation, which I am not privy to the details of), when they could've just announced they've forked Vault internally and kept the code private (sans customers requesting access). Indeed, it is also risky as, if the fork is too successful, it ultimately leaves them competing with themselves, at their own expense. Thus I'm personally hopeful that the end result, given time, will be positive.

Similarly, OpenTofu's success puts pressure to re-open Terraform. But, that community is much stronger than this one, with many people paid to work on it full time as a result of existing competitors in the space building on top (and wanting to contribute back to) Terraform. The risk we run is that HashiCorp (as an IBM company) concedes to the pressure from OpenTofu but not the pressure from OpenBao. But, if a strong enough community forms here, it may not matter whether they open it or not, as the fork will continue to be viable long-term anyways, even if IBM were to step down from the TSC.

My 2c., views my own.

**Reply by vtrenton** [reply]
>it came from within and not from IBM.

I've heard this statement from quite a few Red Hat folks now. Yes - Sales, Marketing, Engineering and Support are all independent of IBM. The leadership really isn't Mergers tend to integrate leadership and desion makers before product and culture. This was not a merocratic decision by Red Hat employees. It clearly came down from the top and the timing of this after the aquision with IBM is more than a liitle suspicious. Most Red Hat employees I know today are against the restriction on Rocky/Alma and were against the CentOS EOL. So all of these new rules are very suspicious from people who have historically been driven by a Merotocracy. Pre IBM they dealt with the Oracle UBL "freeloader" problem and didn't resort to lisencing tactics. IBM has never been afraid to build and support full proprietary programs which is not historically Red Hat.

> The incentive, IMO, is this community. 

which brings me to this point. Hashicorp is not alone in a bubble here. Redis just did a similar thing - there must be some monitary incentive for these organizations to leave Open Source. Else, they wouldn't be doing it at such an alarming rate. There is a massive bate and switch problem. So given this trend and the simple fact that these companies are unabashidly screwing over the communities around these projects - It makes me think that open source is more of a hinderence for these companies than the community is worth. The community damage has already been done here in the reguards of Hashicorp. So for IBM to try to pick up the pieces here - It would be nothing more than a cost for them.

>Further, IBM (or some portion thereof) must recognize the benefits of the LF and the open source community, as they were the ones that started this fork under the LF in the first place. :-)

This is a very good point - but this was before the aquision of Hashicorp was on the table. It's no secret that Hashicorp is a huge integration point for the OpenShift platform. Most of Hashicorps' solutions from the consul service mesh to Vault and Terraform tie very nicely into Red Hat's existing offerings. I could absolutely see OpenBAO being more of strategical positioning pre aquisition of hashicorp. IBM has wanted Vault for quite a while now for a variety of positioning reasons. So IBM could now start changing directions post aquision and likely will reguarding this fork.

It's hard to say what the future holds around this. I really really hope all of this ends up opening back up and remerging. But Hashicorp's recent C&D to OpenTofu sends a very clear message about how friendly they are to these open source forks.

**Reply by cipherboy** [reply]
> IBM has never been afraid to build and support full proprietary programs which is not historically Red Hat.

I do not disagree factually, but will point out that IBM is not a homogeneous organization. Different groups within IBM behave differently and IBM itself sponsors a lot of open source work (whether monetarily through e.g., LF donations, through employee time and contributions, &c). It has sponsored OpenJDK and Linux kernel development, it was one of the founders of the Open Innovation Network, it sponsors the LF, Apache Foundation, the Eclipse Foundation, and no doubt many others.

But one thing I haven't seen, and this may just be the limits of my knowledge, are BUSL licensed code at IBM. :-) If there are any, I'd be curious to hear what they are.

> Redis just did a similar thing - there must be some monitary incentive for these organizations to leave Open Source.

My personal opinion is that shareholder greed drives leadership to make shortsighted changes. When you carefully track employee contributions and their value (ordering changes by expected revenue or some proxy thereof) but fail to do similar accounting for the broader contributions from the community, it causes you to vastly underestimate the value of said community. You only see the cost of your own developer interactions (a negative), added to a zero (omitted) value of the contribution, balancing overall to negative business value. And just counting developer contributions (and ignoring the broader types of contributions--docs improvements, support on forums, speaking at conferences as a user) furthers this under counting. Proper analytics on derived complete value (and yes, costs to support) would be beneficial to any company doing OSS. 

Instead, OSS is seen as an expense, despite arguments from within--as they lack proper data as none has been collected--and thus OSS will be axed as shareholders force leadership to reduce perceived unnecessary costs.

IMO, this explains both the misguided reasoning by companies (in their statements towards the BUSL) and the outrage many feel when this occurs -- they have made many (perhaps small, perhaps large) contributions they have a personal connection to, that they feel the company is writing off and profiting off of, without recognition. 

IIRC, many contributors to Valkey were already contributors to Redis, from large corporations. Their time is not cheap and, from my understanding, had the project familiarity to make large technical changes, when Redis was welcoming. OpenTofu was in a similar position, wherein Terraform was not welcoming of contributions for a period of time, despite people wanting to contribute (and having contributed in the past). But beyond this, these companies advocated for these products, speaking at conferences and helping to grow a community, which again, has positive benefits for the parent project. The broader OSS ecosystem that gets enabled by these projects existing is also important to track and measure (e.g., Vault was used in places like sigstore and  notary project for code signing, or as a backend for database encryption, or as a secrets manager in OpenHorizon), as these reflect well upon the core project from a marketing standpoint.

I will admit, balancing internal roadmap against an unstructured flow of contributions is hard, as a company-owned (not foundation-lead) OSS project. However, publishing some open direction could help this: where we're wanting to go, roughly and subject to change, and areas where we'd be welcoming of contributions from the community. IMO, parts of Red Hat do a good job with this, when they are Red Hat lead projects not owned by an external foundation. Fedora proper is perhaps the best example, as it is a proxy for RHEL direction (ahead of CentOS Stream).

To change this, I think things need to be solved at the board level. Board members need better education about how OSS can work well, how to measure the impacts (both positive and negative) of being OSS, and what impacts relicensing has on a company (certainly HashiCorp and Redis will be case studies in the future). Internally, companies need to do a better job of tracking the full value creation of these ecosystems for the company, so that leadership can have better data to work with. This takes time and connections to change. _Working in Public_ discusses some of this, but not all of it.

Thus, IMHO, the issue is that what we as OSS practitioners understand through practice is interpreted by leadership as subjective opinions, when businesses really need concrete data backing it up that we are not empowered or able to collect.

My 2c.
